app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,22 @@
 class PostsController < ApplicationController
+  layout 'application'
   def index
+    @user = User.find(params[:user_id])
+    @posts = Post.where(user_id: @user.id)
+
+    @posts = @posts.map do |post|
+      comments = post.last_post_comments(post.id)
+      { 'post' => post, 'comments' => comments }
+    end
+
     render 'index'
   end
 
   def show
+    @post = Post.find(params[:id])
+    @comments = @post.last_post_comments(@post.id)
```

Comment:
- [ ] Hi @Abdessittir, based on the requirements of this app, we are supposed to fetch all the comments from the db for a post and display them on the post show page. But it seems you are still filtering the posts. You can formulate a query on the active record using the where clause as such: `@comments = Comment.where(post_id: @post.id)`(just a hint)